src/encoded/ingestion_listener.py

```diff
@@ -0,0 +1,670 @@
+
+        try:
+            with io.BytesIO(manifest_content_formatted.encode('utf-8')) as fp:
+                s3_client.upload_fileobj(fp, Bucket=bundles_bucket, Key=manifest_name, **extra_kwargs)
+
+        except botocore.exceptions.ClientError as e:
+
+            log.error(e)
+
+            message = f"{full_class_name(e)} (while uploading metadata): {str(e)}{additional_info}"
+
+            raise SubmissionFailure(message)
+
+        queue_manager = get_queue_manager(request, override_name=override_name)
+        _, failed = queue_manager.add_uuids([submission_id], ingestion_type=ingestion_type)
+
+        if failed:
+            # If there's a failure, failed will be a list of one problem description since we only submitted one thing.
+            raise SubmissionFailure(failed[0])
+
+    if not success:
+
+        raise SubmissionFailure(message)
+
+    return manifest_content
+
+
+@view_config(route_name='ingestion_status', request_method='GET', permission='index')
+@debug_log
+def ingestion_status(context, request):
+    """ Status route, essentially identical to indexing_status. """
+    ignored(context)
+    queue_manager = request.registry[INGESTION_QUEUE]
+    n_waiting, n_inflight = queue_manager.get_counts()
+    return {
+        'title': 'Ingestion Status',
+        'waiting': n_waiting,
+        'inflight': n_inflight
+    }
+
+
+
+def process_submission(*, submission_id, ingestion_type, app, bundles_bucket=None, s3_client=None):
+    ignored(s3_client)  # we might want to restore the ability to pass this, but no one actually does. -kmp 6-Dec-2021
+    bundles_bucket = bundles_bucket or metadata_bundles_bucket(app.registry)
+    s3_client = make_s3_client()
+    manifest_name = "{id}/manifest.json".format(id=submission_id)
+    log.warning(f'Processing submission {manifest_name}')
+    obj = s3_client.get_object(Bucket=bundles_bucket, Key=manifest_name)
+    # data = json.load(obj)['Body']
+    data = json.load(obj['Body'])
+    email = None
+    try:
+        email = data['email']
+    except KeyError as e:
+        ignored(e)
+        debuglog("Manifest data is missing 'email' field.")
+        if DEBUG_SUBMISSIONS:
+            pass
+            # import pdb; pdb.set_trace()
+    debuglog("processing submission %s with email %s" % (submission_id, email))
+    with vapp_for_email(email=email, app=app) as vapp:
+        if DEBUG_SUBMISSIONS:
+            PRINT("PROCESSING FOR %s" % email)
+        submission = SubmissionFolio(vapp=vapp, ingestion_type=ingestion_type, submission_id=submission_id, log=None)
+        handler = get_ingestion_processor(ingestion_type)
+        result = handler(submission)
+        if DEBUG_SUBMISSIONS:
+            PRINT("DONE PROCESSING FOR %s" % email)
+        return {
+            "result": result,
+            "ingestion_type": ingestion_type,
+            "submission_id": submission_id,
+        }
+
+
+def verify_vcf_file_status_is_not_ingested(request, uuid, *, expected=True):
```

Comment:
This should not be needed as is CGAP specific

Reply:
This implies (yes?) that we don't need `patch_vcf_file_status` which implies we don't need `enqueue_uuids_for_request` which implies we don't need `queue_ingestion` ... but is that last bit right? No need for `queue_ingestion` ?

Reply:
I think probably not. Leave `queue_ingestion` but just remove code paths for the `vcf` stuff, right?